app/main.py

```diff
@@ -1,2 +1,56 @@
 class Dictionary:
-    pass
+
+    INITIAL_CAPACITY = 8
+    LOAD_FACTOR = 2 / 3
+    RESIZE_FACTOR = 2
+
+    def __init__(self):
+        self.elements = [None for _ in range(self.INITIAL_CAPACITY)]
+        self.size = 0
+        self.capacity = self.INITIAL_CAPACITY
+
+    def resize(self):
+        current_elements = [node for node in self.elements if node is not None]
+        self.capacity *= self.RESIZE_FACTOR
+        self.elements = [None for _ in range(self.capacity)]
+        self.size = 0
+        for node in current_elements:
+            self.__setitem__(node[0], node[1])
+
+    def __setitem__(self, key, value):
+
+        if (self.size + 1) > self.capacity * self.LOAD_FACTOR:
+            self.resize()
+
+        hash_ = hash(key)
+        index = hash_ % len(self.elements)
+        node = [key, value, hash_]
+
+        while self.elements[index] is not None:
+            if self.elements[index][0] == key \
+                    and self.elements[index][2] == hash_:
```

Comment:
It is bad to check in that order because, there is no case when `keys` are **equal** but `hashes` **not**